test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -601,6 +601,83 @@ public void WordEmbeddingsTest()
             Done();
         }
 
+        [Fact]
+        public void OnnxTypeConversionTest()
+        {
+            var mlContext = new MLContext(seed: 1);
+            string filePath = GetDataPath("type-conversion.txt");
+
+            // These are the supported conversions
+            // ML.NET does not allow any conversions between signed and unsigned numeric types
+            // Onnx does not seem to support casting a string to any type
+            // Though the onnx docs claim support for byte and sbyte, 
+            // CreateNamedOnnxValue in OnnxUtils.cs throws a NotImplementedException for those two
+            DataKind[,] supportedConversions = new[,]
+            {
+                { DataKind.Int16, DataKind.Int16},
+                { DataKind.Int16, DataKind.Int32},
+                { DataKind.Int16, DataKind.Int64},
+                { DataKind.Int16, DataKind.Single},
+                { DataKind.Int16, DataKind.Double},
+                { DataKind.UInt16, DataKind.UInt16},
+                { DataKind.UInt16, DataKind.UInt32},
+                { DataKind.UInt16, DataKind.UInt64},
+                { DataKind.UInt16, DataKind.Single},
+                { DataKind.UInt16, DataKind.Double},
+                { DataKind.Int32, DataKind.Int16},
+                { DataKind.Int32, DataKind.Int32},
+                { DataKind.Int32, DataKind.Int64},
+                { DataKind.Int32, DataKind.Single},
+                { DataKind.Int32, DataKind.Double},
+                { DataKind.Int64, DataKind.Int16},
+                { DataKind.Int64, DataKind.Int32},
+                { DataKind.Int64, DataKind.Int64},
+                { DataKind.Int64, DataKind.Single},
+                { DataKind.Int64, DataKind.Double},
+                { DataKind.UInt64, DataKind.UInt16},
+                { DataKind.UInt64, DataKind.UInt32},
+                { DataKind.UInt64, DataKind.UInt64},
+                { DataKind.UInt64, DataKind.Single},
+                { DataKind.UInt64, DataKind.Double},
+                { DataKind.Single, DataKind.Single},
+                { DataKind.Single, DataKind.Double},
+                { DataKind.Double, DataKind.Single},
+                { DataKind.Double, DataKind.Double}
+            };
+
+            for (int i = 0; i < supportedConversions.GetLength(0); i++)
+            {
+                var fromKind = supportedConversions[i, 0];
+                var toKind = supportedConversions[i, 1];
+
+                TextLoader.Column[] columns = new []
+                {
+                    new TextLoader.Column("Value", fromKind, 0, 0)
+                };
+                var dataView = mlContext.Data.LoadFromTextFile(filePath, columns);
```

Comment:
Do you need to use a file just to contain a single value `3`? Why not use LoadFromEnumerable, and then you can delete the file?

Reply:
I strictly don't need to save a file. I can generate the file and delete it too.
But in this particular case, the text loader simplifies somethings like automatically converting that number to a range of data types that I can iterate through in my loop by simply changing the data kind.

Reply:
Makes sense. I guess you could do the same with properties of different types, but this works.